dali/operators/image/crop/bbox_crop.cc

```diff
@@ -14,143 +14,725 @@
+either `start` and `end` of the region or `start` and shape` depending on the value of ``bbox_layout``.
+E.g. ``bbox_layout=\"xyXY\"`` means the bounding box coordinates are following the order
+``start_x, start_y, end_x, end_y``, while ``bbox_layout=\"xyWH\" indicates the order is
+``start_x, start_y, width, height``. See more in ``bbox_layout`` argument description.
+
+A second input `labels` can be optionally provided, representing the labels associated with each of
+the bounding boxes.
+
+Outputs: 0:`anchor`, 1:`shape`, 2:`bboxes`, (3:`labels`,)
+
+The resulting crop parameters are provided as two separate outputs `anchor` and
+`shape`, that can be fed directly to `Slice` operator to perform the actual cropping of the
+original image. `anchor` and `shape` contain starting coordinates and dimensions for the crop, as
+`[x, y, (z,)]` and `[w, h, (d,)]` respectively. The coordinates can be represented in absolute or
+relative terms, depending of whether the fixed `crop_shape` was used or not.
+
+The third and fourth output correspond to the adjusted bounding boxes and optionally
+their corresponding labels. Bounding boxes are always specified in relative coordinates.)code")
+    .NumInput(1, 2)  // [boxes, labels (optional),]
+    .InputDox(
+        0, "boxes", "2D TensorList of float",
+        "Relative coordinates of the bounding boxes represented as a 2D tensor where the first "
+        "dimension refers to the index of the bounding box and the second dimension refers to the "
+        "index of the coordinate.")
+    .InputDox(1, "labels", "1D TensorList of integers",
+              "(optional) Labels associated with each of the bounding boxes.")
+    .NumOutput(3)  // [anchor, shape, bboxes, labels (optional),]
+    .AdditionalOutputsFn([](const OpSpec &spec) {
+      return spec.NumInput() - 1;  // +1 if labels are provided
+    })
     .AddOptionalArg(
         "thresholds",
-        R"code(Minimum overlap (Intersection over union) of the bounding boxes with respect to the prospective crop.
-Selected at random for every sample from provided values. Default imposes no restrictions on Intersection over Union for boxes and crop.)code",
+        R"code(Minimum intersection-over-union (IoU) of the bounding boxes with respect to the
+crop window. Each sample will select one of the `thresholds` randomly, and the operator will
+try a given number of attempts (see `num_attempts`) to produce a random crop window that yields an
+IoU above the selected threshold.)code",
         std::vector<float>{0.f})
-    .AddOptionalArg(
-        "aspect_ratio",
-        R"code(Range `[min, max]` of valid aspect ratio values for new crops. Value for `min` should be greater or equal to `0.0`.
-Default values disallow changes in aspect ratio.)code",
-        std::vector<float>{1.f, 1.f})
+    .AddOptionalArg("aspect_ratio",
+                    R"code(Single or multiple valid aspect ratio ranges
+(``[min0, max0, (min1, max1, min2, max2)]``) for cropping windows.
```

Comment:
How this corresponds to `[min_xy, max_xy, min_xz, max_xz, min_yz, max_yz]`?

Reply:
Exactly in the order you wrote. I adjusted the text for better readability